scripts/find_outliers.py

```diff
@@ -22,6 +26,196 @@ def find_outliers(data_directory):
+        imgfiles[str(k)] = i[1]
+        k+=1
+
+    #print(vol_means)
+    # print(volumes[str(1)].shape)  #making sure this gives the shape of 1 run
+    #plot with subplots the means for each volume for each run
+    """fig, ax = plt.subplots(20,1)
+    for i, ax in enumerate(ax):
+        ax.plot(vol_means[str(i)])
+        ax.set_ylabel('Run ' + str(i))
+        ax.set_xlim(0,162)
+        print(len(vol_means[str(i)]))
+    plt.show()"""
+
+
+    """iqr_proportion = 1.5
+    for line in vol_means:
+        q1, q3 = np.percentile(vol_means[line], [25, 75])
+        iqr = q3 - q1
+        up_thresh = q3 + iqr * iqr_proportion
+        down_thresh = q1 - iqr * iqr_proportion
+        print(np.logical_or(vol_means[line] > up_thresh, vol_means[line] < down_thresh))"""
+
+
+
+
+
+
+    #making a residuals dictionary that contains a more linear version
+    # of the mean data. It gets rid of any drift in the data.
+
+
+    """residuals = {}
+    for i in range(len(vol_means)):
+        data = vol_means[str(i)]
+        shape = data.shape[0]  # number of volumes
+        X = np.ones((shape, 2))  # make a 2 column  vector containing ones
+        drift = np.linspace(-1, 1, shape) # make an array of numbers stretching from -1 to 1
+        X[: , 1] = drift # puts the values of drift into the second column of X
+        beta = np.linalg.pinv(X).dot(data) # numpy function to give values of intercept and slope of best fitting line
+        fitted = X.dot(beta)  #creating the best fitted line
+        residual = data - fitted  #difference between mean array and the fitted array
+        residuals[str(i)] = residual + beta[0]  #adds the intercept to show the fitted version against the original mean value version
+
+        #print(np.corrcoef(drift, residual))
+        #print(beta)
+        #plt.plot(fitted)
+        #plt.show()
+
+        #plt.plot(vol_means[str(6)]) #looking at one of the mean plots
+        #plt.plot(residuals[str(6)]) #looking at one of the residual plots
+        #plt.show()"""
+
+
+#adding a section to select values from ONLY outside of the brain
+#Did this by looking at the values from a bunch of the volumes for different runs,
+#and finding that almost all values inside of (volume_mean + 0.3*mean) were inside the brain
+#we can then run dvars on these values so we are not getting the variablity of voxels inside the brain
+    volumes_outside_brain = {}
+
+    for i in range(len(volumes)):
+        #selecting a given run of volumes
+        run = volumes[str(i)]
+        num_vols = volumes[str(i)].shape[-1]
+        outside_brain = np.zeros(volumes[str(i)].shape)
+        #now looping through each volume in the given run
+        for k in range(num_vols):
+            vol = run[...,k]
+            #lowvals = vol < (np.mean(vol) + 0.3*np.mean(vol))
+            #selected = vol[lowvals]
+            #outside_brain.append(selected)
+            lowvals = stats.threshold(vol, threshmax = np.mean(vol) + 0.3*np.mean(vol), newval = 0)
```

Comment:
I preferred the commented code - it doesn't use an external function, so easier to follow.  Maybe as:

```
outside_mask = vol <= np.mean(vol) * 1.3
outside_brain[outside_mask, k] = vol[outside_mask]
```

It would be nice to be able to try different thresholds, like 1.2, 1.4 without editing the code - another good reason to refactor this big function into a collection of smaller functions.